test/groovy/util/JenkinsReadFileRule.groovy

```diff
@@ -9,12 +9,19 @@ class JenkinsReadFileRule implements TestRule {
 
     final BasePipelineTest testInstance
     final String testRoot
+    final String testFilename
 
     JenkinsReadFileRule(BasePipelineTest testInstance, String testRoot) {
         this.testInstance = testInstance
         this.testRoot = testRoot
     }
 
+    JenkinsReadFileRule(BasePipelineTest testInstance, String testRoot, String testFileName) {
+        this.testInstance = testInstance
```

Comment:
I believe we do not need any change in that rule class here. See other comment below. But in case we assume we need this change: We should forward from the other constructor to the new constructor having one parameter more in order to avoid different handing in both constructors.